polkadot/xcm/src/v5/mod.rs

```diff
@@ -1344,7 +1354,7 @@ impl<Call> TryFrom<OldInstruction<Call>> for Instruction<Call> {
 			HrmpChannelClosing { initiator, sender, recipient } =>
 				Self::HrmpChannelClosing { initiator, sender, recipient },
 			Transact { origin_kind, require_weight_at_most: _, call } =>
-				Self::Transact { origin_kind, call: call.into() },
+				Self::Transact { origin_kind, call: call.into(), fallback_max_weight: None },
```

Comment:
Why not populate this? It is only used if needed. Populating it here allows clean v4->v5->v4 transition.

```suggestion
				Self::Transact { origin_kind, call: call.into(), fallback_max_weight: Some(require_weight_at_most) },
```